Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -330,7 +330,7 @@ GlyphData FontCascade::glyphDataForCharacter(UChar32 c, bool mirror, FontVariant
     if (mirror)
         c = u_charMirror(c);
 
-    auto emojiPolicy = resolveEmojiPolicy(FontVariantEmoji::Normal, c);
+    auto emojiPolicy = resolveEmojiPolicy(m_fontDescription.variantEmoji(), c);
```

Comment:
I'd appreciate it if there was a test (even if it fails) determining whether this works in the complex text codepath.